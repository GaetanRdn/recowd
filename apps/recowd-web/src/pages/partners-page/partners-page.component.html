<rc-partners-list [partners]="_partners"></rc-partners-list>
