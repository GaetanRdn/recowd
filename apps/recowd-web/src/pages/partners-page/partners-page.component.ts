import {
  ChangeDetectionStrategy,
  Component,
  inject,
  OnInit,
} from '@angular/core';
import { PartnersListComponent } from '@recowd/ui/organisms';
import { Observable, take } from 'rxjs';
import { Partner } from '@recowd/models';
import { PartnerService } from '../../services/partner.service';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'rc-partners-page',
  standalone: true,
  imports: [CommonModule, PartnersListComponent],
  templateUrl: './partners-page.component.html',
  styleUrls: ['./partners-page.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PartnersPageComponent implements OnInit {
  protected _partners$!: Observable<Partner[]>;

  private readonly _partnerService: PartnerService = inject(PartnerService);

  public ngOnInit(): void {
    this._partners$ = this._partnerService.getAll().pipe(take(1));
  }
}
