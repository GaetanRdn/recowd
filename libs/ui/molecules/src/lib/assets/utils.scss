@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
