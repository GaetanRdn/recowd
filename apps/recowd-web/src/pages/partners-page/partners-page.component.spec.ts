import { TestBed, waitForAsync } from '@angular/core/testing';

import { PartnersPageComponent } from './partners-page.component';
import { TemplateLookup } from '@recowd/test/utils';

describe('PartnersPageComponent', () => {
  let templateLookup: TemplateLookup<PartnersPageComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [PartnersPageComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(PartnersPageComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());
});
