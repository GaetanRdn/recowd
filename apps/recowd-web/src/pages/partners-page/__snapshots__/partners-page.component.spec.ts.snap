// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnersPageComponent should create 1`] = `
<rc-partners-list>
  <rc-card>
    <div
      class="rc-card-header"
    />
    <rc-card-content
      class="rc-card-content"
    >
      <rc-partner-link>
        <a
          class="rc-button-icon"
          href="https://www.norsys.fr/"
          rcbuttonicon=""
          target="_blank"
        >
          <rc-norsys-custom-icon>
            <rc-custom-icon>
              <img
                alt="norsys"
                height="300"
                src="assets/images/norsys.png"
                width="300"
              />
            </rc-custom-icon>
          </rc-norsys-custom-icon>
          
        </a>
      </rc-partner-link>
    </rc-card-content>
  </rc-card>
  <rc-card>
    <div
      class="rc-card-header"
    />
    <rc-card-content
      class="rc-card-content"
    >
      <rc-partner-link>
        <a
          class="rc-button-icon"
          href="https://www.cbp.fr/"
          rcbuttonicon=""
          target="_blank"
        >
          <rc-cbp-custom-icon>
            <rc-custom-icon>
              <img
                alt="cbp"
                height="300"
                src="assets/images/cbp.svg"
                width="300"
              />
            </rc-custom-icon>
          </rc-cbp-custom-icon>
          
        </a>
      </rc-partner-link>
    </rc-card-content>
  </rc-card>
  
</rc-partners-list>
`;
